atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleAlterTableTest.java

```diff
@@ -0,0 +1,180 @@
+import com.palantir.common.exception.TableMappingNotFoundException;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class OracleAlterTableTest {
+
+    private static final Namespace NAMESPACE = Namespace.create("test_namespace");
+    private static final TableReference TABLE_REFERENCE = TableReference.create(NAMESPACE, "foo");
+
+    private static final DbKeyValueServiceConfig CONFIG = ImmutableDbKeyValueServiceConfig.builder()
+            .from(DbKvsOracleTestSuite.getKvsConfig())
+            .ddl(ImmutableOracleDdlConfig.builder()
+                    .overflowMigrationState(OverflowMigrationState.FINISHED)
+                    .build())
+            .build();
+
+    private static final DbKeyValueServiceConfig CONFIG_WITH_ALTER = ImmutableDbKeyValueServiceConfig.builder()
+            .from(CONFIG)
+            .ddl(ImmutableOracleDdlConfig.builder()
+                    .overflowMigrationState(OverflowMigrationState.FINISHED)
+                    .addAlterTablesOrMetadataToMatch(TABLE_REFERENCE)
+                    .build())
+            .build();
+
+    @ClassRule
+    public static final TestResourceManager TRM =
+            new TestResourceManager(() -> ConnectionManagerAwareDbKvs.create(CONFIG));
+
+    private static final String COLUMN_NAME = "variable";
+
+    private static final TableMetadata EXPECTED_TABLE_METADATA = TableMetadata.builder()
+            .singleRowComponent("name", ValueType.STRING)
+            .singleNamedColumn("var", COLUMN_NAME, ValueType.STRING)
+            .build();
+
+    private static final TableMetadata OLD_TABLE_METADATA = TableMetadata.builder()
+            .singleRowComponent("name", ValueType.STRING)
+            .singleNamedColumn("var", COLUMN_NAME, ValueType.FIXED_LONG)
+            .build();
+
+    private static final Cell DEFAULT_CELL = createCell("foo", "bar");
+    private static final Cell DEFAULT_CELL_2 = createCell("something", "bar");
+
+    private static final byte[] DEFAULT_VALUE = PtBytes.toBytes("amazing");
+    private static final long TIMESTAMP = 1L;
+
+    private ConnectionManagerAwareDbKvs defaultKvs;
+    private ConnectionSupplier connectionSupplier;
+    private OracleTableNameGetter oracleTableNameGetter;
+
+    @Before
+    public void setup() {
+        defaultKvs = (ConnectionManagerAwareDbKvs) TRM.getDefaultKvs();
+        connectionSupplier = DbKvsOracleTestSuite.getConnectionSupplier(defaultKvs);
+        oracleTableNameGetter = OracleTableNameGetterImpl.createDefault((OracleDdlConfig) CONFIG.ddl());
+    }
+
+    @After
+    public void tearDown() {
+        defaultKvs.dropTables(defaultKvs.getAllTableNames());
+        connectionSupplier.close();
+    }
+
+    @Test
+    public void whenConfiguredAlterTableToMatchMetadataAndOldDataIsStillReadable() {
+        defaultKvs.createTable(TABLE_REFERENCE, EXPECTED_TABLE_METADATA.persistToBytes());
```

Comment:
Nice! To make this more readable, add some whitespace between the logical gaps and comment the state of the world at each point :)